### ISSUE TITLE
npm run build fails: cannot find named export from nostr-tools package

### ISSUE THREAD

**Issue by salimp2009** [withastro/astro#6341]
### What version of `astro` are you using?

1.9.0

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Hi, 
I installed nostr-tools package using npm 
https://www.npmjs.com/package/nostr-tools

then I used named imports in an .astro file under /pages 
like below and several others
`import { nip19 } from "nostr-tools";`
when I use 
`npm run dev `; it works
by when I run `npm run build `then I get this error (both during the deploy and also local run)
```
Named export 'nip19' not found. The requested module 'nostr-tools' is a CommonJS module, which may not support all module.exports as named exports.
 CommonJS modules can always be imported via the default export, for example using:
import pkg from 'nostr-tools';
const { nip19, relayInit } = pkg;
```
I also checked npm and nostr-tools source repo , the package exports nip19 and all other ones I used;
https://www.npmjs.com/package/nostr-tools
https://github.com/nbd-wtf/nostr-tools

And even if I use recommended via default export, both npm run dev and build works but language server shows a false error . I was told to add     "allowSyntheticDefaultImports": true to tsconfig.json and I did still get the same false error. 

Module "...." has default export. ( I am using astro-language-server 0.29.6) in nvim

my  source code is also available at 
https://github.com/salimp2009/astro_tutorial/blob/main/src/pages/notes.astro

### Steps to Reproduce

1. `npm create astro@latest` 
2.  `npm i nostr-tools`
3. add `import { nip19 } from "nostr-tools" `in index.astro
4. add a sample  code as below to use nip19
```
const pubKeyhex= nip19.decode(
  "npub1sadasdasdasdadadasd"
).data;
```

### Link to Minimal Reproducible Example

https://github.com/salimp2009/astro_tutorial/blob/main/src/pages/notes.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by salimp2009** [comment]
> Upgrading `notstr-tools` to latest should fix the issue, there was a recent commit that fixes that [nbd-wtf/nostr-tools@04a46b8](https://github.com/nbd-wtf/nostr-tools/commit/04a46b815c76b002b4427ca37d3cf25c8b344d55). It's failing because the package isn't loaded correctly in nodejs. Closing as this is out of scope of Astro.

thank you.

**Comment by RobbieTheWagner** [comment]
How do we tell people to fix this issue? Getting the same problem with https://github.com/nasa8x/rss-to-json
